airbyte-db/src/main/java/io/airbyte/db/jdbc/DefaultJdbcDatabase.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2020 Airbyte
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.db.jdbc;
+
+import io.airbyte.commons.functional.CheckedConsumer;
+import io.airbyte.commons.functional.CheckedFunction;
+import java.io.Closeable;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.stream.Stream;
+import javax.sql.DataSource;
+
+/**
+ * Database object for interacting with a JDBC connection. Can be used for any JDBC compliant db.
+ */
+public class DefaultJdbcDatabase implements JdbcDatabase {
+
+  private final DataSource ds;
+
+  public DefaultJdbcDatabase(final DataSource ds) {
+    this.ds = ds;
+  }
+
+  @Override
+  public void execute(CheckedConsumer<Connection, SQLException> query) throws SQLException {
+    try (final Connection connection = ds.getConnection()) {
+      query.accept(connection);
+    }
+  }
+
+  @Override
+  public <T> Stream<T> resultSetQuery(CheckedFunction<Connection, ResultSet, SQLException> query,
+                                      CheckedFunction<ResultSet, T, SQLException> recordTransform)
+      throws SQLException {
+    try (final Connection connection = ds.getConnection()) {
```

Comment:
ungh. you're right. vestigial from the list.

Reply:
i'm going to add a method to the iface that's `bufferedResultSetQuery` so that a caller is able to make a db call without having to manually close a connection.